jni/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -1148,3 +1148,130 @@ BOOL releaseStreamDataBuffer(JNIEnv* env, jobject dataBuffer, UINT32 offset, PBY
 
     return TRUE;
 }
+
+BOOL setStreamEventMetadata(JNIEnv* env, jobject streamEventMetadata, PStreamEventMetadata pStreamEventMetadata)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    jmethodID methodId = NULL;
+    CHECK(env != NULL && streamEventMetadata != NULL && pStreamEventMetadata != NULL);
+    const char *retChars;
+
+    // Load KinesisVideoFrame
+    jclass cls = env->GetObjectClass(streamEventMetadata);
+    if (cls == NULL) {
+        DLOGE("Failed to create StreamEventMetadata class.");
+        CHK(FALSE, STATUS_INVALID_OPERATION);
+    }
+
+    // Retrieve the methods and call them
+
+    methodId = env->GetMethodID(cls, "getVersion", "()I");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getVersion");
+    } else {
+        pStreamEventMetadata->version = env->CallIntMethod(streamEventMetadata, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+
+    methodId = env->GetMethodID(cls, "getImagePrefix", "()Ljava/lang/String;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getImagePrefix");
+    } else {
+        jstring retString = (jstring) env->CallObjectMethod(streamEventMetadata, methodId);
+        CHK_JVM_EXCEPTION(env);
+
+        if (retString != NULL) {
```

Comment:
Can we add a comment that imagePrefix is nullable in the java side (null is OK for this parameter), so do nothing if it's null